symphony-bdk-core/src/test/java/com/symphony/bdk/core/service/datafeed/impl/OnDiskDatafeedIdRepositoryTest.java

```diff
@@ -0,0 +1,84 @@
+package com.symphony.bdk.core.service.datafeed.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.symphony.bdk.core.config.BdkConfigLoader;
+import com.symphony.bdk.core.config.BdkConfigLoaderTest;
+import com.symphony.bdk.core.config.exception.BdkConfigException;
+import com.symphony.bdk.core.config.model.BdkConfig;
+import com.symphony.bdk.core.config.model.BdkDatafeedConfig;
+import com.symphony.bdk.core.service.datafeed.DatafeedIdRepository;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Optional;
+
+/**
+ * Test class for the {@link OnDiskDatafeedIdRepository}.
+ */
+public class OnDiskDatafeedIdRepositoryTest {
+
+  private BdkConfig bdkConfig;
+  private Path datafeedFile;
+
+  @BeforeEach
+  void setUp(@TempDir Path tempDir) throws BdkConfigException {
+    this.bdkConfig = BdkConfigLoader.loadFromClasspath("/config/config.yaml");
+    BdkDatafeedConfig datafeedConfig = this.bdkConfig.getDatafeed();
+    datafeedConfig.setIdFilePath(tempDir.toString());
+    this.bdkConfig.setDatafeed(datafeedConfig);
+    this.datafeedFile = tempDir.resolve("datafeed.id");
+  }
+
+  @Test
+  void writeDatafeedTest() throws IOException {
```

Comment:
tests are easier to read when steps (Given/When/Then) are separated with line breaks IMHO